File: Target LeetdCode 200/ReverseLinkedListII.java

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }
}

class ReverseLinkedListII {
   public

      ListNode reverseBetween(ListNode head, int left, int right) {
        if(head == null || left == right) return head;

        ListNode curr=head,prev=null;
        while(left>1){
            prev=curr;
            curr=curr.next;
            left--;
            right--;
        }

        ListNode con = prev, tail = curr, third = null;
        while(right>0){
            third = curr.next;
            curr.next = prev;
            prev = curr;
            curr= third;
            right--;
        }

        if(con!=null){
            con.next=prev;
        }else{
            head = prev;
        }
        tail.next  = curr;
        return head;
    }

    public static void main(String[] args) {
        
    }
}

File: Target LeetdCode 200/ReverseLinkedListII.cpp
#include <bits/stdc++.h>
using namespace std;

struct ListNode {
    int val;
    ListNode* next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode* next) : val(x), next(next) {}
};

class Solution {
   public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        stack<int> st;
        ListNode *temp = head, *tmp = nullptr;
        for (int i = 1; i != left; i++) temp = temp->next;
        tmp = temp;
        for (int i = left; i != right; i++) st.push(temp->val), temp = temp->next;
        while (st.size()) {
            tmp->val = st.top();
            st.pop();
            tmp = tmp->next;
        }
        return head;
    }
};

int main() {
}